es/es-server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -357,10 +357,7 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                 // now, put the request settings, so they override templates
                 indexSettingsBuilder.put(request.settings());
                 if (indexSettingsBuilder.get(SETTING_NUMBER_OF_SHARDS) == null) {
-                    deprecationLogger.deprecated("the default number of shards will change from [5] to [1] in 7.0.0; "
-                            + "if you wish to continue using the default of [5] shards, "
-                            + "you must manage this on the create index request or with an index template");
-                    indexSettingsBuilder.put(SETTING_NUMBER_OF_SHARDS, settings.getAsInt(SETTING_NUMBER_OF_SHARDS, 5));
+                    throw new IllegalArgumentException("Number of shards must be supplied");
```

Comment:
We could also remove this https://github.com/crate/crate/blob/21b05aa2e7c203954efadd09c61b423fee337885/es/es-server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java#L169 as CrateDB never allowed `:` inside a table name and thus the `deprecationLogger` completely

Reply:
Currently CrateDB allows creating tables containing `:`:

```
cr> create table "t:bar" (x int);
CREATE OK, 1 row affected  (0.290 sec)
cr> select * from "t:bar";
+---+
| x |
+---+
+---+
SELECT 0 rows in set (0.058 sec)
```

Reply:
ah ok, wasn't aware of that.